### DISCUSSION TITLE
Remove overlap of popup sub-menu padding

### DISCUSSION THREAD

**Discussion by micmalti** [tsujan/Kvantum#1003] [Q&A]
Hi, is there a setting similar to `text.margin.left` but for tweaking the horizontal distance of a sub-menu from its primary menu? In my current setup with Plasma 6.0.5, the left side of the sub-menu is overlapping with the right side of the primary menu, and more so in reverse, as you may see below.

![Screenshot_20240618_141118](https://github.com/user-attachments/assets/1a6fe1ad-9ac0-49cd-af11-9f5f2cfb582f)

![Screenshot_20240902_171039](https://github.com/user-attachments/assets/5cd400fc-69a7-4753-a930-bfc4e7c18434)

**Comment by tsujan** [comment]
That depends on the theme. The theme maker may have set `submenu_overlap` intentionally, or the theme may have a problem.

Please check a theme that comes with Kvantum itself.

**Reply by tsujan** [reply]
Also please note that an overlap is expected with the menus that are opened near the right side of the screen. Are the menus of your screenshots so?

**Reply by micmalti** [reply]
That was only for the second screenshot to force the sub-menu to show up on the left-hand-side. The same issue is present when the menus are opened in the middle of the screen.

![Screenshot_20240902_182603](https://github.com/user-attachments/assets/acd46497-d88e-4747-8a4e-5bd31e18e0d6)

**Reply by tsujan** [reply]
As I suggested, try a theme that comes with Kvantum. And use kvantumpreview → File menu to check submenus.

**Comment by micmalti** [comment]
My theme is set with a `submenu_overlap` of 0. I've tested out a couple of Kvantum themes and the issue persisted.

**Reply by tsujan** [reply]
Plasma X11:

![plasma_x11](https://github.com/user-attachments/assets/a4e3ec72-23b6-4f03-b520-cfa95262f865)

Plasma Wayland:

![plasma_wayland](https://github.com/user-attachments/assets/c901fee3-949e-4242-abe6-8961027badf5)

**Reply by micmalti** [reply]
I'm on X11. Is that Plasma 6?

**Reply by tsujan** [reply]
It's Plasma 6.0.5, which I have but never use.

But Plasma or not Plasma, it wouldn't make a difference. I use Kvantum under X11 and Wayland with LXQt, and all my Qt apps are based on Qt6. So, I would see such an overlap immediately if it showed up.

**Reply by micmalti** [reply]
You're right, because when I'm switched the Application Style over to a non-Kvantum theme, the overlap persisted. So, now that it's clear that Kvantum is not at fault here, any idea as to what might be causing this?

**Reply by tsujan** [reply]
> any idea as to what might be causing this?

I can only have a wild guess, and you shouldn't count on it, but there may be a mixture of old and new programs in your system. If that's the case, wait for your distro upgrades. The latest stable Plasma at the time of writing this comment is 6.1.4.

**Reply by micmalti** [reply]
Thanks. I'll inform you if the present situation changes.
